lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,174 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+struct uk_falloc *generate(void *fa_meta, void *start_zone, unsigned long len, __paddr_t *new_start);
+
+
+UK_TESTCASE(ukfallocbuddy, test_init)
+{
+	// test if the init function works
+	struct uk_falloc *fa;
+
+	fa = malloc(uk_fallocbuddy_size());
+	UK_TEST_EXPECT_SNUM_EQ(uk_fallocbuddy_init(fa), 0);
+	free(fa);
+}
+
+UK_TESTCASE(ukfallocbuddy, test_add_memory)
+{
+	// test if the addmem functionality works
+	void *start_zone = malloc(10000);
+	__sz fa_meta_size;
+	__vaddr_t dm_off;
+	__paddr_t start = (__paddr_t)start_zone;
+	__sz len = 10000;
+	unsigned long pages = len >> PAGE_SHIFT;
+	void *fa_meta = malloc(10000);
```

Comment:
Since we get the size of fallocbuddy metadata at line 35, in `fa_meta_size`, using `uk_fallocbuddy_metadata_size`,
should we use that when allocating memory for `fa_meta`?

Same for the other space allocations for the metadata below.